File: frontend/src/types/history.ts
export type HistoryScore = {
  player_id: string
  player_name: string
  player_order: number
  score: number
  correct_answers: number
  answered_questions: number
}

export type HistorySession = {
  id: string
  mode: 'solo' | 'hotseat'
  total_rounds: number
  round_seconds: number
  created_at: string
  finished_at: string | null
  top_score: number
  winners: string[]
  players: HistoryScore[]
}

File: backend/app/models/history.py
from datetime import datetime

from sqlmodel import SQLModel

from backend.app.models.session import SessionMode


class GameHistoryScore(SQLModel):
    """One player row in a stored session summary."""

    player_id: str
    player_name: str
    player_order: int
    score: int
    correct_answers: int
    answered_questions: int


class GameHistorySession(SQLModel):
    """Public summary of one finished local trivia session."""

    id: str
    mode: SessionMode
    total_rounds: int
    round_seconds: int
    created_at: datetime
    finished_at: datetime | None
    top_score: int
    winners: list[str]
    players: list[GameHistoryScore]
